### ISSUE TITLE
-[NSToolbarFullScreenWindow redispatchKeyEvent:]: unrecognized selector sent to instance 

### ISSUE THREAD

**Issue by paulcbetts** [electron/electron#6630]
Electron 1.2.7, macOS only

Seen a few reports of this:

```
Date/Time:             2016-07-27 17:33:16.936 -0700
OS Version:            Mac OS X 10.11.5 (15F34)
Report Version:        11
Anonymous UUID:        BFDB9318-F4C7-3E88-A2B0-838D74414044

Sleep/Wake UUID:       F6F84732-50DF-4A03-BC5A-005598A95D59

Time Awake Since Boot: 390000 seconds
Time Since Wake:       3500 seconds

System Integrity Protection: enabled

Crashed Thread:        0  CrBrowserMain  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Application Specific Information:
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSToolbarFullScreenWindow redispatchKeyEvent:]: unrecognized selector sent to instance 0x7f9123c3bef0'
terminating with uncaught exception of type NSException
abort() called

Application Specific Backtrace 1:
0   CoreFoundation                      0x00007fff8fa324f2 __exceptionPreprocess + 178
1   libobjc.A.dylib                     0x00007fff88c1af7e objc_exception_throw + 48
2   CoreFoundation                      0x00007fff8fa9c1ad -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
3   CoreFoundation                      0x00007fff8f9a2571 ___forwarding___ + 1009
4   CoreFoundation                      0x00007fff8f9a20f8 _CF_forwarding_prep_0 + 120
5   Electron Framework                  0x000000010ee0779a _ZN4atom25CommonWebContentsDelegate19HandleKeyboardEventEPN7content11WebContentsERKNS1_22NativeWebKeyboardEventE + 202
6   Electron Framework                  0x000000010edce75d _ZN4atom3api11WebContents19HandleKeyboardEventEPN7content11WebContentsERKNS2_22NativeWebKeyboardEventE + 45
7   Electron Framework                  0x000000010edce752 _ZN4atom3api11WebContents19HandleKeyboardEventEPN7content11WebContentsERKNS2_22NativeWebKeyboardEventE + 34
8   Electron Framework                  0x000000010edce76d _ZThn80_N4atom3api11WebContents19HandleKeyboardEventEPN7content11WebContentsERKNS2_22NativeWebKeyboardEventE + 13
9   Electron Framework                  0x000000010f856edb _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 9935307
10  Electron Framework                  0x000000010f856aa7 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 9934231
11  Electron Framework                  0x000000010f85577a _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 9929322
12  Electron Framework                  0x000000010f8555ae _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 9928862
13  Electron Framework                  0x000000010f85522f _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 9927967
14  Electron Framework                  0x000000010f8ade41 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 10291505
15  Electron Framework                  0x000000010f89b3f9 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 10215145
16  Electron Framework                  0x000000010fe7ffb8 _CTFontManagerUnregisterFontForData + 3800696
17  Electron Framework                  0x000000010ef3ebcb _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 399035
18  Electron Framework                  0x000000010ef5f633 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 532771
19  Electron Framework                  0x000000010ef5f94c _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 533564
20  Electron Framework                  0x000000010ef5fb3b _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 534059
21  Electron Framework                  0x000000010ef345f1 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 356577
22  Electron Framework                  0x000000010ef5618a _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 494714
23  Electron Framework                  0x000000010ef33ff4 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 355044
24  CoreFoundation                      0x00007fff8f9c7881 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
25  CoreFoundation                      0x00007fff8f9a6fbc __CFRunLoopDoSources0 + 556
26  CoreFoundation                      0x00007fff8f9a64df __CFRunLoopRun + 927
27  CoreFoundation                      0x00007fff8f9a5ed8 CFRunLoopRunSpecific + 296
28  HIToolbox                           0x00007fff90d87935 RunCurrentEventLoopInMode + 235
29  HIToolbox                           0x00007fff90d8776f ReceiveNextEventCommon + 432
30  HIToolbox                           0x00007fff90d875af _BlockUntilNextEventMatchingListInModeWithFilter + 71
31  AppKit                              0x00007fff8d83bdf6 _DPSNextEvent + 1067
32  AppKit                              0x00007fff8d83b226 -[NSApplication _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 454
33  AppKit                              0x00007fff8d82fd80 -[NSApplication run] + 682
34  Electron Framework                  0x000000010ef34e26 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 358678
35  Electron Framework                  0x000000010ef34444 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 356148
36  Electron Framework                  0x000000010ef740e3 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 617427
37  Electron Framework                  0x000000010f67c4bb _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 7991211
38  Electron Framework                  0x000000010f67c2c3 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 7990707
39  Electron Framework                  0x000000010f67e8d2 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 8000450
40  Electron Framework                  0x000000010f677f5d _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 7973453
41  Electron Framework                  0x000000010f5e9981 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 7390321
42  Electron Framework                  0x000000010f5e8af6 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 7386598
43  Electron Framework                  0x000000010ed75307 AtomMain + 71
44  Slack                               0x000000010ed6eec6 main + 38
45  libdyld.dylib                       0x00007fff8eaf05ad start + 1

Thread 0 Crashed:: CrBrowserMain  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff83e38f06 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff8ae7c4ec pthread_kill + 90
2   libsystem_c.dylib             	0x00007fff8ecff6e7 abort + 129
3   libc++abi.dylib               	0x00007fff88be0f81 abort_message + 257
4   libc++abi.dylib               	0x00007fff88c06a47 default_terminate_handler() + 267
5   libobjc.A.dylib               	0x00007fff88c1d6ae _objc_terminate() + 103
6   libc++abi.dylib               	0x00007fff88c0419e std::__terminate(void (*)()) + 8
7   libc++abi.dylib               	0x00007fff88c03c12 __cxa_throw + 121
8   libobjc.A.dylib               	0x00007fff88c1b08c objc_exception_throw + 318
9   com.apple.CoreFoundation      	0x00007fff8fa9c1ad -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
10  com.apple.CoreFoundation      	0x00007fff8f9a2571 ___forwarding___ + 1009
11  com.apple.CoreFoundation      	0x00007fff8f9a20f8 _CF_forwarding_prep_0 + 120
12  com.github.electron.framework 	0x000000010ee0779a atom::CommonWebContentsDelegate::HandleKeyboardEvent(content::WebContents*, content::NativeWebKeyboardEvent const&) + 202
13  com.github.electron.framework 	0x000000010edce75d atom::api::WebContents::HandleKeyboardEvent(content::WebContents*, content::NativeWebKeyboardEvent const&) + 45
14  com.github.electron.framework 	0x000000010edce752 atom::api::WebContents::HandleKeyboardEvent(content::WebContents*, content::NativeWebKeyboardEvent const&) + 34
15  com.github.electron.framework 	0x000000010edce76d non-virtual thunk to atom::api::WebContents::HandleKeyboardEvent(content::WebContents*, content::NativeWebKeyboardEvent const&) + 13
```

**Comment by kevinsawicki** [comment]
@paulcbetts I pushed a fix for this in #6637 but I had trouble reproducing it locally with full screen windows. Are you able to reproduce it consistently?

**Comment by paulcbetts** [comment]
@kevinsawicki Not consistently but we've gotten enough reports of it that it's A Thing. It indeed does seem to be related to full-screen windows

**Comment by CharlieHess** [comment]
Reopening; after releasing the Mac public beta this is one of our top crash reports. :pensive:

**Comment by kevinsawicki** [comment]
@CharlieHess what version of Electron are you seeing this on? It was released in 1.3.2

**Comment by CharlieHess** [comment]
@kevinsawicki Ah, should have checked the dates more closely. We stayed on 1.2.7 for the beta. We'll let you know if we see it again once we upgrade.
